lib/rmt/cli/mirror.rb

```diff
@@ -22,7 +24,9 @@ def self.mirror(repository_url = nil, local_path = nil)
     repositories.each do |repository|
       begin
         puts "Mirroring repository #{repository.name}"
-        mirror_one_repo(repository.external_url, repository.local_path, repository.auth_token)
+        repo_url = repository.external_url
+        repo_url.sub!(/.*(?=SUSE)/, "file://#{from_dir}/") if from_dir
```

Comment:
Consider looking at `Repository.make_local_path` rather than using a regexp, to have all the local path extraction logic in one place.